modules/openapi-generator/src/main/resources/scala-sttp-client/apiInvoker.mustache

```diff
@@ -0,0 +1,54 @@
+{{>licenseInfo}}
+package {{{mainPackage}}}.core
+
+import java.io.File
+
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+import org.json4s.JsonAST.JString
+import org.json4s._
+import sttp.client._
+import org.openapitools.client.api.EnumsSerializers
+import org.openapitools.client.core.ApiInvoker.DateTimeSerializer
+import sttp.client.json4s.SttpJson4sApi
+import sttp.client.monad.MonadError
+
+class SttpSerializer(implicit val format: Formats = DefaultFormats ++ EnumsSerializers.all + DateTimeSerializer,
+                     implicit val serialization: org.json4s.Serialization = org.json4s.jackson.Serialization) extends SttpJson4sApi
+
+class HttpException(val statusCode: Int, val statusText: String, val message: String) extends Exception(s"[$statusCode] $statusText: $message")
+
+object ApiInvoker {
+
+  /**
+    * Allows request execution without calling apiInvoker.execute(request)
+    * request.result can be used to get a monad wrapped content.
+    *
+    * @param request the apiRequest to be executed
+    */
+  implicit class ApiRequestImprovements[R[_], RE, T](request: RequestT[Identity, Either[ResponseError[Exception], T], Nothing])
```

Comment:
I'd use simplified `ApiRequest` instead, since it's mostly wrapper?

Reply:
`RE` looks unused

Reply:
`ApiRequest` could interfered with model name. As that class implicitly defined it better to have better self-description